file-ingestion-test/src/main/java/com/hazelcast/jet/tests/file/ingestion/FileIngestionTest.java

```diff
@@ -0,0 +1,252 @@
+            for (int i = 0; i < jobTypes.length; i++) {
+                JobType jobType = jobTypes[i];
+                JobConfig jobConfig = new JobConfig();
+                jobConfig.setName(name + "-" + jobType + "-" + jobNumber);
+                jobs[i] = client.newJob(pipeline(jobType, jobNumber), jobConfig);
+            }
+            for (Job job : jobs) {
+                job.join();
+            }
+            for (JobType jobType : jobTypes) {
+                verifyObservable(client.getObservable(observableName(jobType, jobNumber)));
+            }
+            sleepSeconds(sleepSeconds);
+            jobNumber++;
+        }
+    }
+
+    private void verifyObservable(Observable<Long> observable) {
+        Iterator<Long> iterator = observable.iterator();
+        assertTrue(observable.name() + " is empty", iterator.hasNext());
+        assertEquals(ITEM_COUNT, iterator.next().intValue());
+        assertFalse(iterator.hasNext());
+        observable.destroy();
+    }
+
+    @Override
+    protected void teardown(Throwable t) {
+    }
+
+    private Pipeline pipeline(JobType jobType, int jobNumber) {
+        Pipeline p = Pipeline.create();
+
+        p.readFrom(source(jobType))
+                .groupingKey(ignored -> 0)
+                .aggregate(counting())
+                .map(Map.Entry::getValue)
+                .writeTo(Sinks.observable(observableName(jobType, jobNumber)));
+
+        return p;
+    }
+
+    private BatchSource<String> source(JobType jobType) {
+        switch (jobType) {
+            case LOCAL:
+                return FileSources.files(localDirectory).build();
+            case LOCAL_WITH_HADOOP:
+                return FileSources.files(localDirectory).useHadoopForLocalFiles(true).build();
+            case HDFS:
+                return FileSources.files(hdfsPath)
+                        .option("fs.defaultFS", hdfsUri)
+                        .option("fs.hdfs.impl", DistributedFileSystem.class.getName())
+                        .option("fs.file.impl", LocalFileSystem.class.getName())
+                        .build();
+            case S3:
+                return FileSources.files("s3a://" + bucketName + "/" + s3Directory)
+                        .option("fs.defaultFS", hdfsUri)
+                        .option("fs.s3a.access.key", accessKey)
+                        .option("fs.s3a.secret.key", secretKey)
+                        .option("fs.hdfs.impl", DistributedFileSystem.class.getName())
+                        .option("fs.file.impl", LocalFileSystem.class.getName())
+                        .build();
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+
+    private void createSourceFiles(JetInstance client) throws Exception {
+        List<Integer> items = IntStream.range(0, ITEM_COUNT).boxed().collect(toList());
+
+        // clear local directory
+        IOUtil.delete(Paths.get(localDirectory));
```

Comment:
Local directory should be cleaned on remote nodes. I think we should move it to cleaning as part of setup in hazelcast-jet-ansible repo.